docs/source/adding_new_benchmark.rst

```diff
@@ -0,0 +1,180 @@
+Adding a New Benchmark
+######################
+
+This guide explains how to integrate a new benchmark into **metriq-gym**. You'll learn how to:
+
+1. Define the core Python structures (classes and :code:`dataclass` objects).
+2. Create a JSON Schema to validate benchmark parameters.
+3. Provide an example schema for convenience.
+4. Register the new benchmark within the package so it becomes accessible to the rest of the system.
+
+Defining a New Benchmark
+************************
+
+1. **Create a New Python File**
+
+   In the :file:`benchmarks` directory of the :code:`metriq-gym` package, add a new file named
+   :file:`benchmarks/<NEW_BENCHMARK>.py` (replace :code:`<NEW_BENCHMARK>` with a descriptive name).
+
+2. **Implement the Benchmark Class**
+
+   In this new file, define a class that inherits from :code:`Benchmark`. You must override:
+
+   - :code:`dispatch_handler()`: Houses the logic for dispatching and scheduling the benchmark job to the quantum device.
+   - :code:`poll_handler()`: Houses the logic for retrieving and processing results from the quantum device or simulator.
+
+3. **Create the Data Classes**
+
+   Define two :code:`dataclass` objects: one to store the input parameters (inheriting from :code:`BenchmarkData`), and one
+   to hold the benchmark’s output results (inheriting from :code:`BenchmarkResult`).
+
+   Example:
+
+   .. code-block:: python
+
+       from dataclasses import dataclass
+       from metriq_gym.benchmarks.benchmark import (
+           Benchmark,
+           BenchmarkData,
+           BenchmarkResult,
+       )
+
+       @dataclass
+       class NewBenchmarkResult(BenchmarkResult):
+           """Stores the result(s) from running New Benchmark."""
+           pass
+
+       @dataclass
+       class NewBenchmarkData(BenchmarkData):
+           """Stores the input parameters or metadata for New Benchmark."""
+           pass
+
+       class NewBenchmark(Benchmark):
+           """Benchmark class for New Benchmark experiments."""
+
+           def dispatch_handler(
+               self,
+               device: QuantumDevice
+           ) -> NewBenchmarkData:
+               # TODO: Implement logic for dispatching the job
+               pass
+
+           def poll_handler(
+               self,
+               job_data: BenchmarkData,
+               result_data: list[GateModelResultData]
+           ) -> NewBenchmarkResult:
+               # TODO: Implement logic for retrieving and processing results
+               pass
+
+Defining the Schema
+*******************
+
+To standardize and validate the input parameters for each benchmark, **metriq-gym** uses JSON Schema. Add a new
+file named :file:`new_benchmark.schema.json` to the :file:`schemas/` directory:
```

Comment:
Make it clear that what follows is an example of a schema

Reply:
Sounds good, I've added the following clause to make this more clear:

```
 Note that this schema file is just an example and should be modified to fit the specific requirements of your benchmark.
```